ocaml/xapi/xapi_vm_lifecycle.ml

```diff
@@ -150,15 +150,16 @@ let has_feature ~vmgmr ~feature =
  *  to perform an operation. This makes a difference for ops that require the guest to
  *  react helpfully. *)
 let check_op_for_feature ~__context ~vmr ~vmgmr ~power_state ~op ~ref ~strict =
-  if power_state <> `Running ||
-     (* PV guests offer support implicitly *)
-     not (Helpers.has_booted_hvm_of_record ~__context vmr) ||
-     Xapi_pv_driver_version.(has_pv_drivers (of_guest_metrics vmgmr)) (* Full PV drivers imply all features *)
-  then None
+  let some_err e = Some (e, [ Ref.string_of ref ]) in
+
+  if power_state <> `Running then None
```

Comment:
I think we need to leave `not (Helpers.has_booted_hvm_of_record ~__context vmr) || Xapi_pv_driver_version.(has_pv_drivers (of_guest_metrics vmgmr))` in the condition